docs/design-system.md

```diff
@@ -0,0 +1,73 @@
+- Button variants in figma are square and rectangle. The additional variants of circle and icon are added outside of figma because there are occasions in the editor and dashboard where there are no shared button styles but the element is expected to behave as a button. By having a button variant of icon we can eliminate the need of having specific exportable buttons like `ArrowDown`.
+- Button types, sizes, and variants (circle, rectangle, square, icon) are all available as constants and importable via `BUTTON_TYPES`, `BUTTON_SIZES`, `BUTTON_VARIANTS` so that we don’t have to worry about miscellaneous strings.
+- Every button combination has a [demo in storybook](https://google.github.io/web-stories-wp/storybook/?path=/story/designsystem-components-button--default).
+
+1. Typography
+
+- [design-system/typography](https://github.com/google/web-stories-wp/tree/main/assets/src/design-system/components/typography) has the only nested content. This follows the foundation set up in figma of Display, Headline, Label, Paragraph, Link.
+- Each typography type has a set of presets that are stored in the theme and matching constants ([theme/constants/typography](https://github.com/google/web-stories-wp/blob/main/assets/src/design-system/theme/constants/typography.js)).
+- The typography components are `<Display />`, `<Headline />`, `<Link />`, `<Text />`
+- Each is a styled component and can be passed whatever element you want it to be with `as` or `forwardedAs` depending on your situation.
+- These are meant to decrease our need to individually style text elements and ensure consistency.
+- There’s a demo for every [type setting in storybook](https://google.github.io/web-stories-wp/storybook/?path=/story/designsystem-components-typography-display--default).
+
+1. Circular dependencies
+
+- If you are adding a component and you need to reference anything in the design-system you’ll need to import it from the nearest location to that import not the top level index.
+- While this works just fine in the app (because you’re outside of the design system), storybook will yell at you because it thinks that imports aren’t loaded in time.
+- As an example, I’m creating this new `<Taco />` and I need a `<Text />` component. I’d need to `import { Text } from ‘../typography’;` not `../` if I’m in `components/taco`.
+
+## Icons
+
+Icons are separated from components and already exported as `ReactComponent` . You can import them via `/theme/icons` or `/theme`.
+
+### How to add icons
+
+Icons are separated into sections to make traversing the directory a little easier when you’re unsure of what all is in there. [Material Design icon categories](https://material.io/resources/icons/?style=baseline) were used as the naming convention so as not to reinvent the wheel.
+
+1. Export your SVG & compress it (example: [https://jakearchibald.github.io/svgomg/](https://jakearchibald.github.io/svgomg/))
```

Comment:
FYI: Manual compression like this is a good habit, but not a requirement. SVGs are optimized during webpack bundling using [SVGO](https://github.com/svg/svgo)